aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -169,8 +182,17 @@ SparseTensor& div_out_sparse_zerodim(SparseTensor& r, const SparseTensor& t, con
   AT_ASSERT(t.is_sparse());
 
   if (is_same_tensor(r, t)) {
+    // Can't divide an uncoalesced integral tensor accurately. e.g. for a sparse int tensor with value 6
+    // represented as values=[3, 3], integer division by 2 would give values=[1, 1] => 2 instead
+    // of 6 / 2 => 3
```

Comment:
This fixes an existing bug unrelated to the main issue. I could move it to a separate stack, but it seems like I get the best test coverage by including the tests added in this PR.

Reply:
If it is not too hard to separate this, I would prefer to review the two things separately (type promotion support vs fixing uncoalesced sparse division) to speed up review time.

Reply:
hmm... I'm trying to avoid testing the same code twice with multiple overlapping tests. Maybe I could just extract the test from this PR without the mixed-type combinations though to include in the first diff.

Reply:
Why can't you stack one of the diffs on top of the other, so that both of them get access to the tests?

Reply:
that's what I mean. I just have to make sure in the first diff that it doesn't test any of the mixed-type ops in this diff but I think that should work.